x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/analysis/analyzer/Verifier.java

```diff
@@ -588,6 +588,33 @@ private static void validateInExpression(LogicalPlan p, Set<Failure> localFailur
                 In.class));
     }
 
+    private static void validateConditional(LogicalPlan p, Set<Failure> localFailures) {
+        p.forEachExpressions(e ->
+            e.forEachUp((ConditionalFunction cf) -> {
+                    int idx = 0;
+                    DataType dt = DataType.NULL;
+
+                    // Find the data type of the first non-null argument
+                    for (; idx < cf.children().size(); idx++) {
+                        DataType childDT = cf.children().get(idx).dataType();
+                        if (childDT != DataType.NULL) {
```

Comment:
I opt for `Expressions.isNull` as it might catch null literals with a different type than `NULL`.